model/SyncLog/SyncLogService.php

```diff
@@ -70,11 +70,14 @@ private function getStorage() {
      * Create new record in storage implementation.
      *
      * @param SyncLogEntity $entity
-     * @return integer Created record ID
+     * @return SyncLogEntity
      */
     public function create(SyncLogEntity $entity)
     {
-        return $this->getStorage()->create($entity);
+        $entityId = $this->getStorage()->create($entity);
+        $this->setEntityId($entity, $entityId);
```

Comment:
Exception is not handled here.

Reply:
It was intentional. These exceptions are handled in client code which uses this service (event listeners in this case), since it can be handles in a different way